omise/controllers/front/payment.php

```diff
@@ -0,0 +1,28 @@
+<?php
+if (! defined('_PS_VERSION_')) {
+    exit();
+}
+
+if (defined('_PS_MODULE_DIR_')) {
+    require_once _PS_MODULE_DIR_ . '/omise/classes/charge.php';
+}
+
```

Comment:
@nimid I agree about separate classes idea.
Btw from this design, in fact, the dependencies weren't removed. It just moves from `OmisePaymentModuleFrontController` to `Charge`.

What I meant is, we can do more decoupling by, let `Controller` take cares only `input`. And then, `Charge` take cares "How the class gonna communicate to Omise API and return back to controller" (i.e. amount formatting.)

For example
Controller just handle "input" and pass those input to `Charge`
```php
// OmisePaymentModuleFrontController.php

public function initContent()
{
    parent::initContent();

    try {
        $charge = new Charge;
        $charge->create(array(
            'amount'      => $this->context->cart->orderTotal(),
            'card'        => Tools::getValue('omise_card_token'),
            'currency'    => $this->context->currency->iso_code,
            'description' => "...",
        ));
    } catch (Exception $e) {
        $this->context->smarty->assign('error_message', $e->getMessage());
    }

    $this->setTemplate('payment_result.tpl');
}
```

And Charge just care about formatting stuff, and how to communicate to Omise API
```php
// Charge.php

public function create($params)
{
    $setting = new Setting();

    switch ($setting->payment_action) {
        case 'manual_capture':
            return $this->authorize($params, $setting->getSecretKey());
            break;

        case 'auto_capture':
            return $this->capture($params, $setting->getSecretKey());
            break;
    }

    return // fail, unknow command.
}

public function authorize($params, $secret_key)
{
    $params['amount']  = OmisePluginHelperCharge::amount($params['currency'], $params['amount']);
    $params['capture'] = false;

    return OmiseCharge::create($params, '', $secret_key);
}

public function capture($params, $secret_key)
{
    $params['amount']  = OmisePluginHelperCharge::amount($params['currency'], $params['amount']);
    $params['capture'] = true;

    return OmiseCharge::create($params, '', $secret_key);
}
```

From this design, let says, if we have "payment_action" == "internet banking" (just assumed we're gonna store "internet banking" option into `payment_action`

We just have another method called `internetBanking() { ... }` and then, this method will do stuff to communicate with Omise API.